# V1: Splitting Train and Test Sets

## 2.1 Creating splits for V1 / Saving Train and Test Sets

This notebook:

1. Loads the cleaned dataset
2. Creates the train/test split
3. Saves the train and test split into artifacts for V1 and V2

In [ ]:
from sklearn.model_selection import train_test_split
from pathlib import Path
import pandas as pd

df = pd.read_csv('data/processed/cleaned_v1.csv')
X, y = df['diff'], df['top_level_label']

X_train_raw, X_test_raw, y_train, y_test = train_test_split(X, y, test_size = .2, random_state = 42, stratify=y)

Path("data/processed").mkdir(parents=True, exist_ok=True)
train = pd.concat([X_train_raw, y_train], axis=1).reset_index(drop=True)
test = pd.concat([X_test_raw, y_test], axis=1).reset_index(drop=True)

train.to_csv('data/processed/train_v1.csv', index=False)
test.to_csv('data/processed/test.csv', index=False)


## 2.2 Verifying Shape and Empty Rows

In [ ]:
assert X_train_raw.index.equals(y_train.index)
assert X_test_raw.index.equals(y_test.index)

print(f'Train Shape: {train.shape}')
print(f'Test Shape: {test.shape}')
print(f'Train Columns: {train.columns}')
print(f'Test Columns: {test.columns}\n')
print('Train Set:')
print(train.isna().sum())
print('\n')
print('Test Set:')
print(test.isna().sum())

Train Shape: (28512, 2)
Test Shape: (7129, 2)
Train Columns: Index(['diff', 'top_level_label'], dtype='str')
Test Columns: Index(['diff', 'top_level_label'], dtype='str')
Train Set:
diff               0
top_level_label    0
dtype: int64
Test Set:
diff               0
top_level_label    0
dtype: int64


## Conclusion:

The cleaned dataset was stratified training training and test sets containing 28,512 examples and 7129 examples, respectively. Both datasets contain the expected columns and contain no missing values.

The resulting datasets were saved to `data/processed/train_v1.csv` and `data/processed/test.csv`